CIP-0062/README.md

```diff
@@ -0,0 +1,215 @@
+
+These definitions extend [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/) to provide specific support for vote delegation.
+
+# **Motivation**
+The goal for this CIP is to extend the dApp-Wallet web bridge to enable the construction of transactions containing metadata that conforms to
+[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format - Updated)](https://cips.cardano.org/cips/cip36/),
+enabling new functionality including vote delegation to either private or public representatives (dReps),
+splitting or combining of private votes,
+the use of different voting keys or delegations
+for different purposes (Catalyst etc).
+
+# **Specification**
+
+## `Types`
+
+### **GovernanceKey**
+
+```
+type GovernanceKey = {
+  votingKey: string,
+  weight: number
+}
+
+```
+
+`votingKey`: Ed25519 pubkey 32 bytes HEX string  
+
+`weight`: Used to calculate the actual voting power using the rules described
+in 
+[CIP-36](https://cips.cardano.org/cips/cip36/).
+
+
+### **Purpose**
+
+```
+type enum Purpose = {
+  CATALYST = 0,
+  OTHER = 1
+}
+
+```
+
+`Purpose`: Defines the purpose of the delegations. This is used to limit the scope of the delegations.  For example, a purpose might be a subset of Catalyst proposals, a council election, or even some private purpose (agreed by convention).
+
+## **Namespace**
+
+### **cardano.{walletName}.governance.enable(): Promise\<API>**
+The `cardano.{walletName}.governance.enable()` method is used to enable the governance API. It should request permission from the wallet to enable the API. If permission is granted, the rest of the API will be available. The wallet should maintain a specific whitelist of allowed clients for this API. This whitelist can be used to avoid asking for permission every time.
+
+This api being an extension of [CIP-30 (Cardano dApp-Wallet Web Bridge)](https://cips.cardano.org/cips/cip30/), expects that `cardano.{walletName}.enable()` to be enabled and added to CIP-30 whitelist implicitly.
+
+When both this API and **CIP-30** is being enabled, is up to the wallet to decide the number of prompts requesting permissions to be displayed to the user.
+
+# **`Jormungandr API`**
+
+## **api.submitVotes**(votes: Vote[], spendingCounter: number): Promise\<hash32>
+
+### `spendingCounter`: 
+The spending counter is used to prevent double voting. The current spending counter for the account should be provided and the implementation should increment it for each vote before submission and attach it to the vote according to [Jormungandr Voting] (https://input-output-hk.github.io/jormungandr/jcli/vote.html#voting). This needs to be in sequential order.
+
+### **`votes`**: 
+
+```
+interface Vote {
+  proposal: Proposal,
+  choice: number,
+  expiration: BlockDate
+}
+```
+#### `expiration`: 
+chain epoch \& slot for when the vote will expire. The type used is: 
+
+```
+interface BlockDate {
+    epoch: number
+    slot: number
+}
+```
+
```

Comment:
How does the dApp calculate the expiration BlockDate?  How does it know the current Jormungandr BlockDate in order to calculate an appropriate expiry?